### DISCUSSION TITLE
Are there dropdown components?

### DISCUSSION THREAD

**Discussion by starascendin** [skeletonlabs/skeleton#1560] [Q&A]
Hi, i'm a n00b to the stack of svelte, tailwind, and skeleton, is there a component for dropdown? I've looked all over but couldn't find anything close to it.

Am i missing it, or it's actually missing from the framework?

Thanks!

**Comment by endigo9740** [comment]
@starascendin depends on what you need, but we do provide styling for native inputs, such as the Select input:
https://www.skeleton.dev/elements/forms#select

Additionally we provide the Popup utility for generating your own custom popup menus, including drop downs (what we use for the Explore and Theme dropdowns top-right of our site):
https://www.skeleton.dev/utilities/popups

We also provide some guided instruction for generating a Combobox - a custom selection drop down:
https://www.skeleton.dev/utilities/popups#combobox

**Comment by cognominal** [comment]
As seen below, It is pretty verbose and scary for the newcomer.  I understand that comes with bells and whitles but most people want menus that looks like system menus. Factorizing it provide a default no-frill drop down menu for the newcomer would be great.  Making it look like the system menus on a given platform would be great.
Also for people with a shopping list, not seing an example with drop down menu will make them choose another toolkit.

https://github.com/skeletonlabs/skeleton/blob/99754539d3950baab77046650954f824ebb95b6f/sites/skeleton.dev/src/lib/components/DocsAppBar/DocsAppBar.svelte

**Reply by endigo9740** [reply]
> As seen below, It is pretty verbose and scary for the newcomer. 

Not quite sure what you're referring to - the popup menu feature and Floating UI? That's why we provide documentation for how to use them. That's your instruction for how to setup and utilize the feature. This is linked my prior message above.

> I understand that comes with bells and whitles but most people want menus that looks like system menus. Factorizing it provide a default no-frill drop down menu for the newcomer would be great. Making it look like the system menus on a given platform would be great.

Not sure what you're after here. The closest that comes to mind is the native Select menu:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select

Skeleton provides opt-in styles for form elements like this, which means you have a choice between our styles or default native styles - that's for you to decide. But it's still the native system powering the element.

The new Popover API will provide more options in this regard, but it's not yet standard across all browsers:
https://developer.mozilla.org/en-US/docs/Web/API/Popover_API

> Also for people with a shopping list, not seing an example with drop down menu will make them choose another toolkit.

Again, I don't really know how to help you here. Folks can use Skeleton if it meets their demands, simple as that. As mentioned in the previous post (back in May) we provide a variety of options to cover number of use cases. If Skeleton doesn't tick a box for you, then it's open source and we would welcome feature requests and/or direct contributions to update, expand, and improve the library.

Given the age of this post you responded to, it's probably worth nothing that quite a lot of effort is going into the next major version of Skeleton right now. So it's a prime time to provide feedback and suggestions:

- https://github.com/skeletonlabs/skeleton/discussions/2375

**Comment by saurabhg772244** [comment]
Is there something similar to [this](https://codesandbox.io/p/sandbox/wispy-paper-nw7ddt?file=%2Fsrc%2FDropdownMenu.tsx%3A67%2C37) in skeleton UI for svelte

**Reply by endigo9740** [reply]
@saurabhg772244 yes, but the answer differ per which version you're using. We're aiming to drop the release candidate for Skeleton v3 this week, so we're recommending most folks move to that when possible. If you have a legacy project, you can make use of the v2 popup feature:

- https://www.skeleton.dev/utilities/popups

In v3 you would use the Popover feature:

- https://next.skeleton.dev/docs/integrations/popover/svelte#popover

In either case you would need to nest these to create a sub-menu as displayed. We tend to favor primitives rather than providing and building a complete implementation like that.
